doc/tutorial/intro-to-autoinstall.rst

```diff
@@ -0,0 +1,250 @@
+ * You can designate particular sections in the autoinstall configuration as
+   "interactive", which means the installer will still stop and ask about those.
+
+
+Providing the autoinstall configuration
+=======================================
+
+There are 2 ways to provide the autoinstall configuration:
+ * Carried as part of cloud-config
+ * Directly on the install media
+
+Autoinstall by way of cloud-config
+----------------------------------
+
+The suggested way of providing autoinstall config to the Ubuntu installer is
+via cloud-init. This allows the configuration to be applied to the installer
+without having to modify the install media.
+
+The autoinstall config is provided via cloud-init configuration, which is
+almost endlessly flexible. In most scenarios the easiest way will be to provide
+user data via the :external+cloud-init:ref:`datasource_nocloud` data source.
+
+When providing autoinstall via cloud-init, the autoinstall config is provided
+as :external+cloud-init:ref:`user_data_formats-cloud_config`. This
+means we need a :code:`#cloud-config` header. The autoinstall directives are
+placed under a top level :code:`autoinstall` key, like so:
+
+.. code-block:: yaml
+
+    #cloud-config
+    autoinstall:
+        version: 1
+        ....
+
+.. note::
+
+   :external+cloud-init:ref:`user_data_formats-cloud_config` files must contain
+   the ``#cloud-config`` header to be recognized as a valid cloud config data
+   file.
+
+Autoinstall on the install media
+--------------------------------
+
+Another option for supplying autoinstall to the Ubuntu installer is to place a
+file named :code:`autoinstall.yaml` on the install media itself.
+
+There are two potential locations for the :code:`autoinstall.yaml` file:
+ * At the root of the "cdrom". When you write the installation ISO to a USB
+   Flash Drive, this can be done by copying the :code:`autoinstall.yaml` to the
+   partition containing the contents of the ISO - i.e.,
+   in the directory containing the ``casper`` sub-directory.
+ * On the rootfs of the installation system - this option will typically
+   require modifying the installation ISO and is not suggested, but is
+   supported.
+
+Directly specifying autoinstall as a :code:`autoinstall.yaml` file does not
+require a :code:`#cloud-config` header, and does not use a top level
+:code:`autoinstall` key. The autoinstall directives are placed at the top
+level.
+
+.. code-block:: yaml
+
+    version: 1
+    ....
+
+
+Cloud-init and autoinstall interaction
+======================================
+
+While cloud-init may provide the autoinstall configuration to the
+Ubuntu installer, it does not process the autoinstall itself.
+
+If cloud-init directives are intended to modify the ephemeral system, they
+must appear at the top level of the cloud-config. If instead
+cloud-init directives are intended to modify the system being installed, they
+must appear under a :code:`user-data` section in :code:`autoinstall`.
+
```

Comment:
Good intro, I think this section is where we get most rub. Let's add a bit more context around where cloud-init runs relative to the ephemeral and installer environment here as cloud-init will point that this section too.

```suggestion
Cloud-init runs in both the ephemeral system, during install, and in the target system during first boot. Cloud-init then becomes inert for every subsequent
reboot.

While cloud-init may provide the autoinstall configuration to the
Ubuntu installer, it does not process the autoinstall directives
itself.

To modify the ephemeral system with cloud-init, any :external+cloud-init:ref:`#cloud-config module schema keys<modules>` can
be provided. If instead cloud-init directives are intended to modify the system
being installed, they must appear under a :code:`user-data:` section under
:code:`autoinstall:`.
```